documentation/homework/Homework4-MobX/my-app/src/store/TodoStore.js

```diff
@@ -0,0 +1,23 @@
+import { observable, computed } from 'mobx'
+import Todo from '../models/Todo'
+
+  class TodoStore {
+      @observable todos = []
+      @observable filter = ''
+
+      @computed
+      get filterList() {
+          const match = new RegExp(this.filter, 'i')
+          return this.todos.filter(todo => !this.filter || match.test(todo.text))
+      }
+
+      createTodo(value) {
+          this.todos.push(new Todo(value))
```

Comment:
I am not sure what is happening here, because you are passing the `value` to a constructor which is same as this store and then you are pushing it to the todos array.

I don't know if this may help but you are initiating a new instance there and a new one here from it value, but it won't work, does it work with you?